src/util/guid.cpp

```diff
@@ -0,0 +1,45 @@
+// Luanti
+// SPDX-License-Identifier: LGPL-2.1-or-later
+// Copyright (C) 2024 SFENCE
+
+#include "guid.h"
+#include <cstring>
+#include <sstream>
+#include <string_view>
+
+#include "exceptions.h"
+#include "util/base64.h"
+#include "log.h"
+
+std::string MyGUID::base64() const
+{
+	return base64_encode(std::string_view(&bytes[0], bytes.size()));
+}
+
+void MyGUID::serialize(std::ostream &os) const
+{
+	os.write(&bytes[0], bytes.size());
+}
+
+void MyGUID::deSerialize(std::istream &is)
+{
+	is.read(&bytes[0], bytes.size());
+}
+
+GUIDGenerator::GUIDGenerator() :
+	m_uniform(0, UINT64_MAX)
+{
+	std::random_device rd;
+	m_rand.seed(rd());
```

Comment:
we're seeding the RNG with a single `unsigned int` here, this is not good.
my suggestion:
```cpp
{
	u64 seed;
	if (!porting::secure_rand_fill_buf(&seed, sizeof(seed))) {
		// main.cpp initializes our internal RNG as good as possible, so fall back to it
		myrand_bytes(&seed, sizeof(seed));
	}

	// Make sure we're not losing entropy or providing too few
	static_assert(sizeof(seed) == sizeof(decltype(m_rand)::result_type), "seed type mismatch");
	m_rand.seed(seed);
}
```

(add `<type_traits>`, `"porting.h"` and `"util/numeric.h"`)